codec/src/main/java/io/netty/handler/codec/ByteToMessageDecoder.java

```diff
@@ -285,7 +286,15 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                 try {
                     if (cumulation != null && !cumulation.isReadable()) {
                         numReads = 0;
-                        cumulation.release();
```

Comment:
Can a check for **cumulation.refCount() >0** be placed here? This will ensure that only those bytebuf are released whose refCount>0 and if bytebuf already has a refcount of 0, it is already deallocated or can be returned to the object pool. This will prevent IllegalReferenceCount from occurring. 
Not sure about the implication on memory (Whether this check will cause a memory leak ?)